modules/generator/registry/registry.go

```diff
@@ -132,7 +134,10 @@ func (r *ManagedRegistry) registerMetric(m metric) {
 	r.metricsMtx.Lock()
 	defer r.metricsMtx.Unlock()
 
-	r.metrics = append(r.metrics, m)
+	if _, ok := r.metrics[m.Name()]; ok {
+		level.Debug(r.logger).Log("msg", "replacing metric, counters will be reset", "metric", m.Name())
```

Comment:
nit: i would make this info level. it won't happen often and it would be good to have a record of it when it does.